### PR TITLE
Smart contracts workspace improvements

### PR DESCRIPTION
**Pull request by magiodev**
## 1. Overview

In this pull request, we've made several enhancements and refinements to the smart contracts workspace to improve its overall functionality and maintainability.

## 2. Implementation details

- **Code Quality Improvements:**
  - Resolved various warnings to enhance the code quality and stability of the workspace (Commit: e7c0020).
  - Polished existing tests to ensure their reliability and coverage (Commit: 1610f31).

- **Build System Updates:**
  - Introduced `justfile` to streamline the build process (Commit: b93839b).

- **Codebase Modernization:**
  - Updated serialization methods: Replaced `to_binary` with `to_json_binary` and `from_json` with `from_json_binary` to align with current standards (Commit: 639b4e3).
  - Fixed and replaced deprecated serialization functions to enhance maintainability (Commits: fd2f635, db5eae6).

- **Dependency Management:**
  - Added more dependencies to the workspace to ensure consistency and ease of management (Commit: a588eb2).
  - Updated `prost` dependencies across all contracts and included them in the workspace configuration to maintain compatibility and performance (Commits: 3f10156, e7eb9b1, 22e73ca).
  - Bumped additional dependencies to their latest versions and incorporated them into the workspace configuration for improved functionality (Commit: 535f190).

- **CI/CD and Build Script Adjustments:**
  - Removed the make command, updated CI to use the just command, and eliminated unused root directories to streamline the build and deployment process (Commit: 80a2e04).

- **Code Reversions and Fixes:**
  - Conducted further polishing and reverted specific changes in the `cl-vault` related to the autocompound feature to align with the intended functionality (Commits: 5614754, c60d73b).

- **Miscellaneous:**
  - Merged the latest changes from the main branch to ensure that the workspace is up-to-date with the base code (Commit: 7d62a7d).
  - A final note indicating a successful return to a stable state (Commit: c60d73b).

## 4. Checklist

- [ ] Does the Readme need to be updated?

### CODE REVIEW

File: smart-contracts/justfile [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+lint:
+	cargo +nightly clippy --all-targets -- -D warnings
+
+gen: build gen-schema
+
+gen-schema:
+	./scripts/schema.sh
+
+integration-test: deploy-local workspace-optimize
```
Comment (magiodev) [inline smart-contracts/justfile:19]: Remove this command we do not use and check all the other ones for similar cases

**Review by LaurensKubat — DISMISSED** [review]
lgtm

**Review by LaurensKubat — APPROVED** [review]
lgtm

**Review by LaurensKubat — APPROVED** [review]
lgtm

**Pull request merged**